src/rpcq.lisp

```diff
@@ -573,38 +282,95 @@ We distinguish between the following options for any field type:
                                     ;; else initialize to
                                     ;; initform generated by TRANSLATE-FIELD-TYPE
                                     `(:initform ,initform))))))
+           (slot-initialization-deprecation-warnings (field-spec)
+             (let* ((slot-name (car field-spec))
+                    (slot-keyword (intern (symbol-name slot-name) :keyword))
+                    (field-settings (cdr field-spec)))
+               (alexandria:when-let* ((deprecated-by (getf field-settings :deprecated-by))
+                                      (deprecated-by-keyword (intern (symbol-name deprecated-by) :keyword)))
+                 `((when (and (getf initargs ,slot-keyword)
+                              (not (getf initargs ,deprecated-by-keyword)))
+                     (warn ,(format nil "~a has been deprecated by ~a."
+                                    slot-keyword deprecated-by-keyword)))))))
+           (slot-accessor-deprecation-warnings (field-spec)
+             (let* ((slot-name (car field-spec))
+                    (slot-keyword (intern (symbol-name slot-name) :keyword))
+                    (field-settings (cdr field-spec)))
+               (alexandria:when-let* ((deprecated-by (getf field-settings :deprecated-by))
+                                      (deprecated-by-keyword (intern (symbol-name deprecated-by) :keyword)))
+                 `((defmethod ,(accessor slot-name) ((obj ,class-name))
```

Comment:
I prefer
```
(list `(defmethod ...))
```
I try to use quasiquote for things that are actually "quoted data", whereas I use `list` and friends for reasons like splicing and mapcan